### PR TITLE
Inline and delete StringUtil#bytesToString [HZ-3484]

### PR DESCRIPTION
**Pull request by orcunc**
Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/util/StringUtil.java [hunk 1]
```diff
@@ -57,28 +57,11 @@ public final class StringUtil {
     public static final Pattern VERSION_PATTERN
             = Pattern.compile("^(\\d+)\\.(\\d+)(\\.(\\d+))?(-\\w+(?:-\\d+)?)?(-SNAPSHOT)?$");
 
-    /**
-     * Empty String.
-     */
-    public static final String EMPTY_STRING = "";
```
Comment (JackPGreen) [inline hazelcast/src/main/java/com/hazelcast/internal/util/StringUtil.java:63]: Why is this removed?

Comment (orcunc) [reply hazelcast/src/main/java/com/hazelcast/internal/util/StringUtil.java:63]: It is dead code.

Comment (arodionov) [reply hazelcast/src/main/java/com/hazelcast/internal/util/StringUtil.java:63]: It is not used

File: hazelcast/src/main/java/com/hazelcast/internal/util/StringUtil.java [hunk 2]
```diff
@@ -57,28 +57,11 @@ public final class StringUtil {
...
 
-    /**
-     * Creates a UTF8_CHARSET string from a byte array.
-     *
-     * @param bytes  the byte array.
-     * @param offset the index of the first byte to decode
-     * @param length the number of bytes to decode
-     * @return the string created from the byte array.
-     */
-    public static String bytesToString(byte[] bytes, int offset, int length) {
```
Comment (arodionov) [inline hazelcast/src/main/java/com/hazelcast/internal/util/StringUtil.java:78]: What about `public static String bytesToString(byte[] bytes)`?
Are you planning to remove it either? 

In a code base, we have a mix of usage - `bytesToString(...)` and  `new String(...)` used both.

Comment (orcunc) [reply hazelcast/src/main/java/com/hazelcast/internal/util/StringUtil.java:78]: I will try to remove that one too. But in a new PR to keep it small. All methods that delegate to JDK code can be removed in my opinion

File: hazelcast/src/main/java/com/hazelcast/internal/util/StringUtil.java (hunk 1 above)
Comment (JackPGreen) [reply hazelcast/src/main/java/com/hazelcast/internal/util/StringUtil.java:63]: > It is dead code.

> It is not used

I'm pretty sure I'm actually using it in a branch, but I guess I'm the only one!

**Review by JackPGreen — APPROVED** [review]

**Review by arodionov — APPROVED** [review]
